### ISSUE TITLE
StringScanner:  undefined method `captures'

### ISSUE THREAD

**Issue by HoneyryderChuck** [oracle/truffleruby#2520]
How to reproduce:

```ruby
irb(main):001:0> require "strscan"
=> true
irb(main):002:0>
irb(main):002:0> s = StringScanner.new('This is an example string')
=> #<StringScanner 0/25 @ "This ...">
irb(main):003:0> s.scan(/\w+/)
=> "This"
irb(main):004:0> s.captures
Traceback (most recent call last):
        from /Users/tiagocardoso/.rubies/truffleruby-21.1.0/bin/irb:42:in `<main>'
        from <internal:core> core/kernel.rb:400:in `load'
        from <internal:core> core/kernel.rb:400:in `load'
        from /Users/tiagocardoso/.rubies/truffleruby-21.1.0/lib/gems/gems/irb-1.3.3/exe/irb:11:in `<top (required)>'
        from <internal:core> core/throw_catch.rb:36:in `catch'
        from <internal:core> core/throw_catch.rb:36:in `catch'
        from <internal:core> core/kernel.rb:437:in `loop'
        from <internal:core> core/kernel.rb:325:in `eval'
(irb):4:in `<top (required)>': undefined method `captures' for #<StringScanner 4/25 "This" @ " is a..."> (NoMethodError)
irb(main):005:0> RUBY_VERSION
=> "2.7.2"
```

`StringScanner#captures` exists at least since ruby 2.4 . I don't know exactly why it isn't there, as I thought this part of the stdlib was shared with truffleruby, but I might be wrong.

**Comment by eregon** [comment]
`strscan` is a pure-Ruby stdlib for TruffleRuby: https://github.com/oracle/truffleruby/blob/master/lib/truffle/strscan.rb (it's also a default gem)
That's probably faster, has less indirections than going through a C extension to match Regexps, and some of the Regexp C-API used by that extension might not be supported yet.

@HoneyryderChuck Would you like to make a PR to add `StringScanner#captures`?
If not I can probably do it.
Either way I can add a check that StringScanner methods match using `spec/truffle/methods_spec.rb`, or that we aware of missing methods.

**Comment by HoneyryderChuck** [comment]
Makes sense, I honestly thought it was using the CRuby C extension version, but I guess this version is more JIT able.

I can't commit this time with a patch, sadly.

**Comment by eregon** [comment]
Fixed in 417feca9a4e32e303fc8cc9429b31a9fab08c4ca.
